ipv-api/src/test/java/uk/gov/di/authentication/ipv/contract/IpvTokenTest.java

```diff
@@ -275,9 +214,70 @@ void getIPVTokenErrorResponse(MockServer mockServer) {
         assertThat(tokenResponse.toHTTPResponse().getStatusCode(), equalTo(400));
     }
 
+    @Disabled
+    @Pact(consumer = "OrchTokenConsumer")
+    RequestResponsePact emptyRequestToGenerateSignedJwt(PactDslWithProvider builder) {
+        return builder.uponReceiving("Valid auth code")
+                .path("/" + IPV_TOKEN_PATH)
+                .method("POST")
+                .matchHeader("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
+                .willRespondWith()
+                .status(200)
+                .body(
+                        newJsonBody(
+                                        (body) -> {
+                                            body.stringType(ACCESS_TOKEN_FIELD, ACCESS_TOKEN_VALUE);
+                                            body.stringType(TOKEN_TYPE_FIELD, TOKEN_TYPE_VALUE);
+                                            body.integerType(EXPIRES_IN_FIELD, EXPIRES_IN_VALUE);
+                                        })
+                                .build())
+                .toPact();
+    }
+
+    @Disabled
+    @Test
+    @PactTestFor(
+            providerName = "IpvCoreBackTokenProvider",
+            pactMethod = "emptyRequestToGenerateSignedJwt",
+            pactVersion = PactSpecVersion.V3)
+    void generationOfJwtSignature(MockServer mockServer) throws ParseException, JOSEException {
+        when(configService.getIPVBackendURI()).thenReturn(URI.create(mockServer.getUrl()));
+        signJWTWithKMS();
+
+        TokenResponse tokenResponse;
+        try (MockedStatic<NowHelper> mockedNowHelperClass = Mockito.mockStatic(NowHelper.class)) {
+            mockedNowHelperClass
+                    .when(NowHelper::now)
+                    .thenReturn(Date.from(Instant.parse("2000-01-01T00:00:00.00Z")));
+            mockedNowHelperClass
+                    .when(() -> NowHelper.nowPlus(5L, ChronoUnit.MINUTES))
+                    .thenReturn(Date.from(Instant.parse("2099-01-01T00:00:00.00Z")));
+            try (var mockJwtId =
+                    mockConstruction(
+                            JWTID.class,
+                            (mock, context) -> {
+                                when(mock.getValue()).thenReturn("1");
+                            })) {
+                tokenResponse = ipvTokenService.getToken("dummyAuthCode");
+            }
+        }
+
+        assertThat(tokenResponse.indicatesSuccess(), equalTo(true));
+    }
+
     private SignResponse mockKmsReturn() {
+        byte[] signature_bytes;
+
+        try {
+            signature_bytes =
+                    ECDSA.transcodeSignatureToDER(
+                            Base64.getDecoder().decode(CLIENT_ASSERTION_SIGNATURE));
```

Comment:
Nice, this make much more sense. At this point we probably don't need the complexity of the disabled methods, we can just use jwt.io